qutip/_mkl/spsolve.py

```diff
@@ -127,95 +136,127 @@ def __init__(self, np_pt=None, dim=None, is_complex=None,
+            print('Solution time:                  ',
+                  round(self._solve_time, 4))
+            print('Solution memory (Mb):           ',
+                  round(self._iparm[16]/1024, 4))
+            print('Number of iterative refinements:',
+                  self._iparm[6])
+            print('Total memory (Mb):              ',
+                  round(sum(self._iparm[15:17])/1024, 4))
             print()
-        
-        # Return solution vector x
-        if nrhs==1:
-            if x.shape != b_shp:
-                x = np.reshape(x, b_shp)
-            return x
-        else:
-            return np.reshape(x, b_shp, order='F')
-    
+        return np.reshape(x, b_shp, order=('C' if nrhs == 1 else 'F'))
+
     def info(self):
         info = {'FactorTime': self._factor_time,
                 'SolveTime': self._solve_time,
-                'Factormem': round(self._iparm[15]/1024.,4), 
-                'Solvemem': round(self._iparm[16]/1024.,4),
+                'Factormem': round(self._iparm[15]/1024, 4),
+                'Solvemem': round(self._iparm[16]/1024, 4),
                 'IterRefine': self._iparm[6]}
         return info
-        
-    
+
     def delete(self):
-        #Delete all data
-        error = np.zeros(1,dtype=np.int32)
-        np_error = error.ctypes.data_as(ctypeslib.ndpointer(np.int32, ndim=1, flags='C'))
-        pardiso(self._np_pt, byref(c_int(1)), byref(c_int(1)), byref(c_int(self._mtype)),
-            byref(c_int(-1)), byref(c_int(self._dim)), self._data, self._indptr, self._indices, 
-            self._np_perm, byref(c_int(1)), self._np_iparm, byref(c_int(0)), byref(c_int(0)),
-            byref(c_int(0)), np_error)
+        # Delete all data
+        error = np.zeros(1, dtype=np.int32)
+        np_error = error.ctypes.data_as(ndpointer(np.int32, ndim=1, flags='C'))
+        pardiso(
+            self._np_pt,
+            byref(c_int(1)),
+            byref(c_int(1)),
+            byref(c_int(self._mtype)),
+            byref(c_int(-1)),
+            byref(c_int(self._dim)),
+            self._data,
+            self._indptr,
+            self._indices,
+            self._np_perm,
+            byref(c_int(1)),
+            self._np_iparm,
+            byref(c_int(0)),
+            byref(c_int(0)),
+            byref(c_int(0)),
+            np_error,
+        )
         if error[0] == -10:
             raise Exception('Error freeing solver memory')
-        
-        
-        
+
+
+_MATRIX_TYPE_NAMES = {
+    4: 'Complex Hermitian positive-definite',
+    -4: 'Complex Hermitian indefinite',
+    2: 'Real symmetric positive-definite',
+    -2: 'Real symmetric indefinite',
+    11: 'Real non-symmetric',
+    13: 'Complex non-symmetric',
+}
+
+
+def _mkl_matrix_type(dtype, solver_args):
+    mtype = (
+        (0b0100 if dtype == np.complex128 else 0b0010)
+        | (0b0000 if solver_args['hermitian'] else 0b1001)
+    )
```

Comment:
Why use bytes and not int?
```
    mtype = (
        (4 if dtype == np.complex128 else 2)
        + (0 if solver_args['hermitian'] else 9)
    )
```

Reply:
I had (erroneously, it turns out) assumed that this was interpreted as a bit field.  It's actually just specific numbers.  I'll change it to be more legible.